packages/amplify-cli-core/src/utils/recursiveOmit.ts

```diff
@@ -0,0 +1,21 @@
+import _ from 'lodash';
+import { $TSObject } from '..';
+
+export const recursiveOmit = (obj: $TSObject, path: Array<string>): void => {
```

Comment:
Totally agree, but this function was moved from: https://github.com/aws-amplify/amplify-cli/blob/eb9257eaee117d0ed53ebc23aa28ecd7b7510fa1/packages/amplify-cli-core/src/deploymentSecretsHelper.ts#L27

I think adding tests or changing behavior is out of scope, because this PR is large enough as is.